src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPDocumentSymbolFeature.java

```diff
@@ -97,29 +96,21 @@ public StructureViewTreeElement getStructureViewTreeElement(@NotNull DocumentSym
     }
 
     /**
-     * Finds the text offset of the symbol as the start of its selection range. Only types and methods/functions are
-     * reported as having a text offset.
+     * Finds the text offset of the symbol as the start of its selection range if available and range if not.
      *
      * @param documentSymbol the document symbol
      * @param psiFile        the file
      * @return the start of the symbol's selection range if valid; otherwise 0
      */
     public int getTextOffset(@NotNull DocumentSymbol documentSymbol,
                              @NotNull PsiFile psiFile) {
-        SymbolKind symbolKind = documentSymbol.getKind();
-        // TODO: This works quite well without having to implement MethodNavigationOffsetProvider
-        if ((symbolKind == SymbolKind.Class) ||
-                (symbolKind == SymbolKind.Interface) ||
-                (symbolKind == SymbolKind.Enum) ||
-                (symbolKind == SymbolKind.Struct) ||
-                (symbolKind == SymbolKind.Method) ||
-                (symbolKind == SymbolKind.Function)) {
-            Range selectionRange = documentSymbol.getSelectionRange();
-            Position startPosition = selectionRange != null ? selectionRange.getStart() : null;
-            Document document = LSPIJUtils.getDocument(psiFile.getVirtualFile());
-            if ((startPosition != null) && (document != null)) {
-                return LSPIJUtils.toOffset(startPosition, document);
-            }
+        // NOTE: This works quite well without having to implement MethodNavigationOffsetProvider

```

Comment:
I noticed while testing that some things were resolving to the top of the file when they should be resolving to the beginning of the declaration. The issue was the specificity in the original logic. Just as in `PsiElement`, `getTextOffset()` should return the start offset of the name identifier if possible, and if not, the start offset of the text range. Now this does for all document symbols.